src/main/java/com/alipay/sofa/common/config/DefaultConfigManager.java

```diff
@@ -17,65 +17,116 @@
 package com.alipay.sofa.common.config;
 
 import com.alipay.sofa.common.utils.OrderComparator;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.util.concurrent.UncheckedExecutionException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.time.Duration;
 import java.util.List;
 import java.util.concurrent.CopyOnWriteArrayList;
 
 /**
  * @author zhaowang
  * @version : SofaCommonConfig.java, v 0.1 2020年10月20日 8:30 下午 zhaowang Exp $
  */
-public class DefaultConfigManger implements ConfigManager{
+public class DefaultConfigManager implements ConfigManager {
 
-    public static final Logger LOGGER = LoggerFactory.getLogger(DefaultConfigManger.class);
+    public static final Logger LOGGER = LoggerFactory.getLogger(DefaultConfigManager.class);
 
+    private final Object EMPTY = new Object();
     private final List<ConfigSource> configSources = new CopyOnWriteArrayList<>();
     private final List<ManagementListener> configListeners = new CopyOnWriteArrayList<>();
+    private LoadingCache<ConfigKey, Object> cache;
+
+    public DefaultConfigManager(long expireAfterSecond, long maximumSize) {
+        this.cache = CacheBuilder.newBuilder()
+                .expireAfterWrite(Duration.ofSeconds(expireAfterSecond))
+                .maximumSize(maximumSize)
+                .build(new CacheLoader<ConfigKey, Object>() {
+                    @Override
+                    public Object load(ConfigKey key) {
+                        Object config = getConfig(key, null);
+                        if (config == null) {
+                            return EMPTY;
+                        }
+                        return config;
+                    }
+                });
+    }
+
 
     @Override
     public <T> T getOrDefault(ConfigKey<T> key) {
-        return getConfig(key,key.getDefaultValue());
+        return getConfig(key, key.getDefaultValue());
     }
 
     @Override
     public <T> T getOrCustomDefault(ConfigKey<T> key, T customDefault) {
-        return getConfig(key,customDefault);
+        return getConfig(key, customDefault);
+    }
+
+    @Override
+    public <T> T getOrDefaultWithCache(ConfigKey<T> key) {
+        return getConfigWithCache(key, key.getDefaultValue());
+    }
+
+    @Override
+    public <T> T getOrCustomDefaultWithCache(ConfigKey<T> key, T customDefault) {
+        return getConfigWithCache(key, customDefault);
+    }
+
+    private <T> T getConfigWithCache(ConfigKey<T> key, T defaultValue) {
+        Object result = null;
+        try {
+            result = cache.getUnchecked(key);
+        } catch (UncheckedExecutionException e) {
+            Throwable cause = e.getCause();
+            if (cause instanceof RuntimeException) {
```

Comment:
what if cause is not RuntimeException?

Reply:
`load` method in line 50 will not throw exception  except RuntimeException, so `cause` must be RuntimeException.

Reply:
According to the javadoc of `UncheckedExecutionException`, it seems there is the possibility of checked exception in some cases.

> However, it may also wrap a checked exception in some cases.